include/circt/Dialect/Handshake/HandshakeOps.td

```diff
@@ -110,20 +110,20 @@ def FuncOp : Op<Handshake_Dialect, "func", [
   let hasCustomAssemblyFormat = 1;
 }
 
-// InstanceOp
-def InstanceOp : Handshake_Op<"instance", [CallOpInterface]> {
-  let summary = "module instantiate operation";
+// CallOp
+def CallOp : Handshake_Op<"call", [CallOpInterface]> {
+  let summary = "Handshake function instantiation operation";
   let description = [{
-    The `instance` operation represents the instantiation of a module.  This
-	  is similar to a function call, except that different instances of the
-	  same module are guaranteed to have their own distinct state.
-    The instantiated module is encoded as a symbol reference attribute named
-    "module". An instance operation takes a control input as its last argument
+    The `call` operation represents the instantiation of a handshake function.
+    This is similar to a function call, except that different calls of the
+	  same function are guaranteed to have their own distinct state.
```

Comment:
I haven't quite been following this part of the discussion, but I think this could use more elaboration of the use models you're trying to target.  I can see a few things and it's not clear to me what this is:
1) Interaction between sequential code (i.e. running on a processor) and a generated accelerator
2) Interaction between accelerators with handshake interfaces.
3) An accelerator with handshake interfaces invoking sequential code.

The comment about 'state' here is I think the critical one... if you have two 'instances', then it seems there some separate state (appropriate for 2, perhaps), while a 'call' would seem to suggest some stateful access to a single piece of hardware (appropriate for 1, perhaps).  At some level, these could be transformed into one another, but there can be subtleties.  (enforcing sequential access reduces scheduling freedom and might require more buffering.  Alternatively two 'instances' with separate state could be shared on a single hardware pipeline while only duplicating the storage (somehow).

Reply:
The original motivation for this PR was to separate the use of "call" and "instance" such that they don't alias (longer discussion here, sorry for not linking that in the initial PR https://github.com/llvm/circt/pull/2118), and by having both we can distinctly say whether we're instantiating a `handshake.func` or a `hw.module`.

I do agree that this can introduce a bit of confusion - the intention, for now, has solely been for `handshake.call` to instantiate a unique handshake module (2).
An alternative to this PR could be to keep the `handshake.instance` same but then let lowering be defined based on the type of the referenced symbol (`handshake.func` vs. `hw.module`). This would remove the possible ambiguity of (1) and (3) by naming the operation `call`.

Eventually, having a way to specify (1) and (3) seems useful, but I'm not convinced that this needs to be modeled at the level of Handshake IR, since this capability seems more general than something restricted to a dataflow model.